nodelet/pkg/utils/netutils/net_utils.go

```diff
@@ -0,0 +1,125 @@
+package netutils
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/apparentlymart/go-cidr/cidr"
+	"github.com/pkg/errors"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/config"
+	"github.com/platform9/nodelet/nodelet/pkg/utils/constants"
+	"golang.org/x/net/nettest"
+)
+
+type NetImpl struct{}
+
+type NetInterface interface {
+	AddrConv(string, int) (string, error)
+	IpForHttp(string) (string, error)
+	GetRoutedNetworkInterFace() (string, error)
+	GetIPv4ForInterfaceName(string) (string, error)
+	GetNodeIP() (string, error)
+	GetNodeIdentifier(config.Config) (string, error)
+}
+
+func New() NetInterface {
+	return &NetImpl{}
+}
+
+//AddrConv generates <pos> th IP from hostCIDR
+func (u *NetImpl) AddrConv(hostCIDR string, pos int) (string, error) {
```

Comment:
```suggestion
func (n *NetImpl) AddrConv(hostCIDR string, pos int) (string, error) {
```
applies to all functions